src/app/videos/[id]/page.tsx

```diff
@@ -1,30 +1,72 @@
 "use client"
+
 import { useEffect, useState } from "react"
-import { useRouter } from "next/navigation"
-import { usePathname, useSearchParams } from "next/navigation"
+import { usePathname } from "next/navigation"
 import Image from "next/image"
+import {
+  FaThumbsUp,
+  FaThumbsDown,
+  FaBookmark,
+  FaRegBookmark,
+} from "react-icons/fa"
 import YoutubeLogo from "/public/images/logos/YouTube-White-Dark-Background-Logo.wine.svg"
 import { formatDate, DateFormats } from "@utils/static/formatDate"
 import { VideoItem } from "types/interfaces/VideoItem"
 import { YoutubeService } from "@/services/YoutubeService"
+import { ref, getDatabase, get } from "firebase/database"
+import firebaseApp from "@config/firebase"
+import fetchLikeCount from "@utils/static/fetchLikeCountVideos"
+import likeVideo from "@utils/static/likeVideos"
+import { toggleBookmark } from "@utils/static/bookmarkItems"
+import useAuth from "@/hooks/useAuth"
+
+const db = getDatabase(firebaseApp)
 
 const VideoPage = () => {
-  const router = useRouter()
+  const { user,loading } = useAuth()
   const pathname = usePathname()
-  const searchParams = useSearchParams()
-
   const idSegment = pathname.split("/").pop()
-  const id = idSegment || searchParams.get("id") || ""
+  const id = idSegment || ""
 
   const [video, setVideo] = useState<VideoItem | null>(null)
   const youtubeService = new YoutubeService()
+  const [likeCount, setLikeCount] = useState<number>(0)
+  const [likeStatus, setLikeStatus] = useState<{ [key: string]: boolean }>({})
+  const [dislikeCount, setDislikeCount] = useState<number>(0)
+  const [dislikeStatus, setDislikeStatus] = useState<{
+    [key: string]: boolean
```

Comment:
There is also a hook lib for firebase so you can create a snapshot and all changes to the FB will automatically set your states and calling setState() will update the FB
https://www.npmjs.com/package/react-firebase-hooks

It is a bit outdated, try it in a clean commit and drop it if you see it can't help you